tests/combine_samples.C

```diff
@@ -0,0 +1,186 @@
+#include <vector>
+#include "mpi.h"
+#include <string>
+#include <stdio.h>
+
+int main(int argc, char* argv[])
+{
+    // Initialize MPI
+    MPI_Init(&argc, &argv);
+    int size;
+    MPI_Comm_size(MPI_COMM_WORLD, &size);
+    int rank;
+    MPI_Comm_rank(MPI_COMM_WORLD, &rank);
+	
+    std::vector<std::string> sample_names;
+    int snaps = 0; 
+    int dim   = 0;
+    bool subtract_mean = false;
+    std::string kind = "snapshot";
+	
+    if (argc >= 2) {
+        for (int i = 1; i < argc; i++) {
+	    if (!strcmp(argv[i], "basis") || !strcmp(argv[i], "-b")) { 
+            if (rank==0) std::cout << "Argument " << i << " identified as basis or -b" << std::endl;
+                kind = "basis"; 
+            } 
+            else if (!strcmp(argv[i], "mean") || !strcmp(argv[i], "-m")) {
+                if (rank==0) std::cout << "Will subtract mean" << std::endl;
+	    	subtract_mean = true;
+	    }
+	    else {
+                sample_names.push_back(argv[i]); 
+	    }
+	}
+    }
+    else {
+        if (rank==0) std::cout << "No arguments passed." << std::endl;
+        return 1;
+    }
+ 
+    /*-- Read dimension and count number of snapshots/bases --*/
+    if (rank==0) std::cout << "Opening files to read dimension and count number of snapshots/bases" << std::endl;
+    for (const auto& sample_name: sample_names) {
+        CAROM::BasisReader reader(sample_name);
+		
+        if (kind == "snapshot") {
+	    CAROM::Matrix *snapshots = (CAROM::Matrix*) reader.getSnapshotMatrix(0);
+	    dim    = snapshots->numRows();
+ 	    snaps += snapshots->numColumns();
+        }
+	else {
+            CAROM::Matrix *basis = (CAROM::Matrix*) reader.getSpatialBasis(0);
+	    dim    = basis->numRows();
+	    snaps += basis->numColumns();
+	}
+    }
+	
+    CAROM_VERIFY((snaps > 0) && (dim > 0));
+	
+    /*-- Load data from input files --*/
+    std::string generator_filename = "total";
+    std::unique_ptr<CAROM::BasisGenerator> static_basis_generator;
+    static_basis_generator.reset(new CAROM::BasisGenerator(
+                                     CAROM::Options(dim, snaps).setMaxBasisDimension(snaps), false,
+                                     generator_filename));
+
+    if (rank==0) std::cout << "Loading data from " << kind << std::endl;
+    for(const auto& sample_name: sample_names) {
+        static_basis_generator->loadSamples(sample_name, kind);
+    }
+
+    if (rank==0) std::cout << "Saving data uploaded as a snapshot" << std::endl;
+    static_basis_generator->writeSnapshot();
+	
+    if (!subtract_mean) {
+        /*-- Compute SVD and save file --*/
+        if (rank==0) std::cout << "Computing SVD" << std::endl;
+        int rom_dim = static_basis_generator->getSpatialBasis()->numColumns();
+        if (rank==0) std::cout << "U ROM Dimension: " << rom_dim << std::endl;
+        static_basis_generator->endSamples();
+    }
+    else {
+        // Close file so it can be opened:
+        static_basis_generator = nullptr;		
```

Comment:
Instead of writing and reading the same data, can you just use `static_basis_generator->getSnapshotMatrix`?

Reply:
Yes! I will change that.